### ISSUE TITLE
The small counter on the wire icon indicating the amount of received messages is not being reset properly

### ISSUE THREAD

**Issue by rbieb** [wireapp/wire-ios#685]
- Have an unread message
- The wire icon on the homescreen shows a litte "1" indicating that I received a message
- Go into the app
- Read the message
- Leave the app again
- Problem: The counter still shows a "1"

A lot of times, I have to actually enter the app twice in order to reset the notification count, it usually doesn't work the first time.

**Comment by MadEgg** [comment]
Same here.  And probably also in relation to #662 - it seems that sometimes Wire gets a push notification that there are new messages, and the first response is to show a pop up that there a new message and to increase the new message counter. This even seems to happen when the new message has not been actually received yet by Wire.

When opening the app because the counter shows new messages and the status screen shows that a message was received, I don't see any new messages. When I wait a while and check back, there's a new message.

So there seems to be some desynchronization in the process of push messages, message reception and the message counter.

IMO the app should not update the counter before it actually has to show something for it, and make sure it's reset to 0 when you've seen all the messages.

**Comment by anatoli26** [comment]
Same here. The counter sometimes is not updated, no indication on the icon that I got a new message (i.e. it stays without any counter).

**Comment by jenskasemets** [comment]
Hi @rbieb @MadEgg @anatoli26 ,

Thank you for your report. Would you be so kind and give us more information when and in which conditions it happens. 
- Which network conditions did you had? Did you change network during this process
- Which device (and iOS version) you have?

**Comment by anatoli26** [comment]
@jenskasemets, I don't remember the network state, these days I'll try to collect as much details as possible and report here.

Do you communicate with the APNS servers directly or via some 3rd-party provider? Just got some feeling... :)

With respect to the versions:
 - iPhone 6S
 - iOS 10.2.1
 - Wire 2.29.2772

**Comment by rbieb** [comment]
@jenskasemets 
 
- iPhone 7
- iOS 10.2.1
- Wire 2.29(2772)

Can't really tell you all that much about my network conditions, but what I know is that it definitely happens on wifi too (5 bars signal strength, all good, only wifi on that channel, quality is pretty much as good as it gets) so I doubt it's related to signal quality.

**Comment by MadEgg** [comment]
For me: iPhone 5S with iOS 10.2.1
Wire version 2.29.2772

I'll pay close attention to circumstances when it happens, but it has happened both on cellular connection and stable wifi. My first guess would be that it happens more frequently if Wire is running in the foreground when a message comes in, but I can't reproduce it at this moment. I'll update when it happens again.

**Comment by MadEgg** [comment]
I found one way that seems to fairly reliably reproduce (part of) the issue.

I always turn my phone off at night. When I turn it on in the morning and messages were sent to me, I do get the notification, but without details: 'New message' rather than 'XXX sent a new message to YYY'. In this case, the counter also does not reflect the new messages.

When starting Wire, the conversation list does show the new messages with the 'dot' before the conversation.

**Comment by anatoli26** [comment]
I'm experiencing the same behavior as described by MadEgg. How to reproduce: put the phone in airplane mode, ask someone to send you a message, turn the airplane mode off so the phone connects to the internet. Then you receive a notification that you have a new message, but the Wire's icon is not updated with the counter (i.e. no counter).

**Comment by anatoli26** [comment]
Also, sometimes there are 2 new messages (2 different contacts with one new message each), Wire gets 2 notifications, but only "1" is shown in the badge.

![img_0267](https://cloud.githubusercontent.com/assets/22968420/23352754/47645712-fcc9-11e6-9e5b-ef07bf8e589d.PNG)

![img_0268](https://cloud.githubusercontent.com/assets/22968420/23352761/4ec8de56-fcc9-11e6-9d24-a08c2bff508b.PNG)

**Comment by jenskasemets** [comment]
@MadEgg  @anatoli26 the behaviour what you describe is actually by design, see explanation in another ticket: https://github.com/wireapp/wire-ios/issues/662#issuecomment-281092234

App icon badge showing "1" is also a limitation of getting fallback notifications — we don't have information about what kind of new messages you have and how many of them are unread. Thus we just show indication that something inside the app needs your attention. 
Alternative option would not send fallback notifications, which means users would start missing new messages.

Let's keep discussion in this ticket on initial issue — badge count not being cleared despite app opened and closed. Looks like it's still an issue for some of you and we would like to fix it, just need more information how to reproduce it.

**Comment by anatoli26** [comment]
@jenskasemets, you're right the original issue here was about resetting the badge counter, but it looks like what is discussed here starting from the 2nd post is that users are receiving new messages (with appropriate notifications) without getting any badge change.

How to reproduce: put the phone in airplane mode, ask someone to send you a message, turn the airplane mode off so the phone connects to the internet. You'll get a new message notification, but there'll be no badge counter change. If there was no badge counter at all, there will still be no badge counter after getting the new message.

I'll comment on the other issue ("1" for 2 unread messages) in the #662 issue.

**Comment by jenskasemets** [comment]
@anatoli26 I'm not I follow your issue. Please create a separate ticket with exact steps to reproduce and what is desired behaviour in your opinion and let's keep this ticket on topic.

**Comment by anatoli26** [comment]
#761

**Comment by MadEgg** [comment]
Okay, another way to reproduce the issue:

* Open a conversation with someone
* Say something
* Close the app before the other party responds (e.g. press the home button)
* Don't use Wire for anything else before the other party responds
* When a response comes in, observe the counter appearing (showing 1)
* Open the received message by either tapping on the message itself, or by tapping on the Wire application icon
* Because the conversation was open when you left Wire, it shows the message directly.
* Close the app by pressing the home button
* Observe that the counter has not been reset even though you read it.
* Open the app again and observe the message again
* Close the app again, and observe that now the counter has been reset


If you switch to a different conversation or to the conversation overview and open Wire afterwards, the counter is reset right away. So it seems that the counter is not reset only when the incoming message is in the conversation you last opened.

**Comment by alexleonard** [comment]
Bump on this. 

Just did some tests with a friend on iOS did the following:

1. Sent her a message in a group chat
2. The Wire icon on her home screen updated and showed a count of (1)
3. Sent her a direct message separate to the group chat
4. The Wire icon continued to show (1).

Initially when she reported the issue I believe she assumed it should show a count of all unread messages across all chats/groups. I guessed that the counter would be 'the number of conversations with new messages'. However it appears neither is the case and it only ever shows a count of 1.

**Comment by rbieb** [comment]
@jenskasemets Any info on this issue since @EgbertW was able to reproduce it?

**Comment by dfarr24** [comment]
I have this issue too. 

if someone sends a new message, the Wire badge icon shows the red 1 that I have an unread message. I open the app, read the message and then close the app by pressing the menu button on my iPhone and the badge icon does not disappear even though I have read the new message. The only way to get it to go away is to either respond to the message or scroll up in the conversation thread and then back down to the now read message. Then the badge icon goes away.

I have iOS 11.4 and Wire 3.16 (3052) and this happens whether connected via wifi or cellular.

This happens every time - I can reproduce reliably.

**Comment by vomout** [comment]
One and a half years and still no word of fix? This makes the app very annoying, having to open and reopen many times to get the badge to reset. (Sometimes pressing on the message seems to help, but maybe it's just my imagination.)

**Comment by milo1892** [comment]
I'm evaluating Wire as my standard messenger app, but this bug is a showstopper for me.

**Comment by nzgrover** [comment]
> 
> 
> One and a half years and still no word of fix? This makes the app very annoying, having to open and reopen many times to get the badge to reset. (Sometimes pressing on the message seems to help, but maybe it's just my imagination.)

Still having this issue. The hint to go and tap on the messages work for me though. Had to "heart" and un-heart two messages to clear the counter. Very annoying.
